theano/gof/vm.py

```diff
@@ -405,8 +405,13 @@ def __call__(self):
         for k in self.storage_map:
             compute_map[k][0] = (k.owner is None)
 
-        # apply_stack contains nodes
-        apply_stack = list(self.base_apply_stack)
+        if output_indices is not None:
+            output_subset = [self.outputs[i] for i in output_indices]
+            apply_stack = [o.owner for o in output_subset]
+        else:
+            # apply_stack contains nodes
+            apply_stack = list(self.base_apply_stack) # sygi: why list?
```

Comment:
Right, but as far as I see the only place where you touch `base_apply_stack` is [here](https://github.com/Theano/Theano/blob/master/theano/gof/vm.py#L333), so it should be a list anyway.